pyproject.toml

```diff
@@ -0,0 +1,102 @@
+[project]
+name = "sktime"
+version = "0.9.0"
+description = "A unified framework for machine learning with time series"
+license = "BSD-3-Clause"
+authors = [
+    {name = "F. Király", email = "f.kiraly@ucl.ac.uk"}
+]
+maintainers = [
+    {name = "F. Király", email = "f.kiraly@ucl.ac.uk"}
+]
+readme = "README.md"
+keywords = [
+    "data-science",
+    "machine-learning",
+    "data-mining",
+    "time-series",
+    "scikit-learn",
+    "forecasting",
+    "time-series-analysis",
+    "time-series-classification",
+    "time-series-regression",
+]
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved",
+    "Programming Language :: Python",
+    "Topic :: Software Development",
+    "Topic :: Scientific/Engineering",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX",
+    "Operating System :: Unix",
+    "Operating System :: MacOS",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+]
+requires-python = ">=3.7,<3.10"
+dependencies = [
+    "deprecated>=1.2.13",
+    "numba>=0.53",
+    "numpy==1.19.3",
+    "pandas>=1.1.0",
+    "scikit-learn>=0.24.0",
+    "statsmodels>=0.12.1",
+]
+
+[project.optional-dependencies]
+all_extras = [
+    "dtw-python",
+    "esig==0.9.7",
+    "fbprophet>=0.7.1",
+    "hcrystalball>=0.1.9",
+    "matplotlib>=3.3.2",
+    "pmdarima>=1.8.0,!=1.8.1",
+    "pyod>=0.8.0",
+    "pystan==2.19.1.1",
+    "scikit_posthocs>=0.6.5",
+    "seaborn>=0.11.0",
+    "stumpy>=1.5.1",
+    "tbats>=1.1.0",
+    "tsfresh>=0.17.0",
+]
+
+dev = [
+    "cython",
+    "pre-commit",
+    "pytest",
+    "pytest-cov",
+    "pytest-xdist",
+    "wheel",
+]
```

Comment:
do we not also require `pydocstyle`, `pycodestyle`, `numpydoc`, and `flake8`? Also, should `numpydoc` not be here too, since it is a `pre-commit` check?

Reply:
`numpydoc` is under `docs` section. All the rest are configured by pre-commit which installs packages configured in `sktime/.pre-commit-config.yaml` first time you run it.